raiden/network/transport/matrix/transport.py

```diff
@@ -775,7 +775,7 @@ def _initialize_first_sync(self) -> None:
             )
             assert len(partner_address) <= 1, msg
             # should contain only one element which is the partner's address
-            if len(partner_address) == 1:
+            if len(partner_address) == 1 and partner_address[0] is not None:
```

Comment:
what guarantees that `None` will always be the first element?

Reply:
Nothing. But that's okay. `partner_address` is a set of addresses derived from the users in the room. There should always be only one address in this set otherwise the room is vulnerable (multiple addresses in the room). If the one partner address is a NoneType then the address could not be recovered from the displayname. This is also a vulnerability because a raiden node should only communicate addresses with a correct displayname. 

An example would be: A server admin could join a room and thus having not the correct displayname. The raiden node should leave that room.